### DISCUSSION TITLE
分享我对Wenda最新版本和Autogpt的看法和不成熟的想法

### DISCUSSION THREAD

**Discussion by shmily91** [wenda-LLM/wenda#175] [Ideas]
这几天都在玩autogpt，今天下了wenda的最新版本，和浏览了闻达的部分代码，作者维护这个项目的精神和更新的速度着实让我感动和惊讶。
我根据我这几天在玩autogpt，以及对作者项目的一些了解，有几个不太成熟的想法想和作者分享以下：
1、在玩autogpt的时候发现autogpt有记忆体，有redis、weaviate，这些可以用docker可以部署在本地电脑，wenda是否有计划部署记忆体。weaviate作为知识图谱，这样会不会更方便些；
2、知识库增强，我只看了bing模式的，流程大概是提取关键词，分别对关键词进行搜索，然后汇总后再让模型回答。当然现在的模式的效果已经非常不错了，我仍有一个不成熟的想法是能否像autogpt一样引入智能体，分解目标步骤，将有用信息存入memory（json、redis、weaviate等），之后智能体验证答案再回答问题。
3、项目增加的autos，让我看到了中国版autogpt的可能性，我浅薄的想法是加入commands功能，中国版autogpt的雏形基本实现。
4、对该项目发展的一点展望：目前autogpt的热度不低，我也挺期待中国版的autogpt的出现，目前我了解的平替项目里，我觉得wenda是最有潜力向这个方面发展的，如果能在人脸识别旁边再引入自动完成任务的功能，或许作为第一个平替项目的autogpt，wenda的热度相信会迎来空前高涨，这或许会吸引更多人的关注和加入。
5、对项目周期的忧虑：基于对该项目的喜爱，我真心希望这个项目周期能够更加持久，这项目的更新速度很快，这让我有一些担忧，担心只有作者一个人在开发，如果只有一个人开发，估计是废寝忘食的状态了。希望项目作者能招募更多的人进行项目的开发，也希望有能力的游戏爱这个项目的人能参与到开发中去。
6、贴个微信二维码吧。这是我浏览github那么久第一次产生支持一个项目的念头，希望表达一下心意。

**Comment by l15y** [comment]
1.之前不知道有weaviate，我甚至在群里问了有没有这种东西。接下来我会抽时间研究，但近期较忙。
2.st模式下使用的方法和你所说类似。sentence transformer做向量提取+faiss做向量存储
3.commands功能没太懂是啥
4.auto下的人脸识别，主要作为开发演示，用户可以自行修改添加功能。做这个的主要目的也是展示一种潜在可能，吸引开发者

**Reply by shmily91** [reply]
> 1.之前不知道有weaviate，我甚至在群里问了有没有这种东西。接下来我会抽时间研究，但近期较忙。 2.st模式下使用的方法和你所说类似。sentence transformer做向量提取+faiss做向量存储 3.commands功能没太懂是啥 4.auto下的人脸识别，主要作为开发演示，用户可以自行修改添加功能。做这个的主要目的也是展示一种潜在可能，吸引开发者

commands功能就是autogpt用来调用控制台窗口的接口，让智能体做判断输入什么命令从而调用控制台，他们好像就是通过这个方式实现智能体对电脑控制，从而通过命令进行网页的搜索和分析、记忆体的使用，甚至是用doker拉去、软件安装、测试等等，通过不断循环使用什么命令和让gp他判断是否实现目标，从而实现autogpt。这是我对autogp比较浅薄的看法，要是有时间你也可以下一下autogpt的代码看看，相信你很快就能弄清楚它的逻辑。看介绍weaviate有六十万篇论文，还有其他数据，用来做知识库说不定是一个比较好的选择。

另外最近了解到有个想做语言模型的stable diffsion的项目，希望作者有时间也了解下，真心希望wenda能在一众项目中脱颖而出

**Reply by alreadydone** [reply]
commands可不只是调用控制台，目前Auto-GPT[已实现了](https://github.com/Significant-Gravitas/Auto-GPT/tree/0efbe23d8902888c5a115c4f0cd8082b9d6a3ba5/autogpt/commands)搜索浏览、发送推文、音频转文字、图像生成、文件操作、代码运行/分析/改进、Git操作等commands，[shell commands](https://github.com/Significant-Gravitas/Auto-GPT/blob/19f893e1e229e57f4a9aa6d7a5acee524e63986e/autogpt/commands/execute_code.py#L102-L111)只是其中一小部分。

> weaviate有六十万篇论文

这意思是做了六十万篇论文的embedding？是什么时候的事？最近Weaviate的合作伙伴Cohere倒是做了百万篇维基百科的embedding：https://txt.cohere.com/embedding-archives-wikipedia/

**Reply by shmily91** [reply]
> commands可不只是调用控制台，目前Auto-GPT[已实现了](https://github.com/Significant-Gravitas/Auto-GPT/tree/0efbe23d8902888c5a115c4f0cd8082b9d6a3ba5/autogpt/commands)搜索浏览、发送推文、音频转文字、图像生成、文件操作、代码运行/分析/改进、Git操作等commands，[shell commands](https://github.com/Significant-Gravitas/Auto-GPT/blob/19f893e1e229e57f4a9aa6d7a5acee524e63986e/autogpt/commands/execute_code.py#L102-L111)只是其中一小部分。
> 
> > weaviate有六十万篇论文
> 
> 这意思是做了六十万篇论文的embedding？是什么时候的事？最近Weaviate的合作伙伴Cohere倒是做了百万篇维基百科的embedding：https://txt.cohere.com/embedding-archives-wikipedia/

搜索浏览用的应该是require 、selenium和google search api；推文有twitter的api接口，图像生成有openai的那个dell（忘了怎么拼了）的api，stable diffison（本地的和huggingface可选）、代码运行、分析、改进估计就是用模型的能力了，只是加了个prompt，音频转文字用了huggingface接口，文件操作这个Git操作用有可能都是用控制台，我感觉大部分功能都写在autogtp的commands是文件夹里。
60w学术论文我看的是这个：https://zhuanlan.zhihu.com/p/416020395  
原来是有人使用 Weaviate 矢量搜索为 60 多万篇学术论文构建可扩展的知识图谱搜索，记错了,不知道那人有没有分享。
我个人猜测是如果平替的模型出来了autogpt的功能时，估计到时有人会训练相对应的模型。所以挺期待wenda能加入这个功能抢占先机，定位可以侧重做知识搜索，相信这会使这个项目的热度更上一层楼，同时或许能够吸引更多的人加入这个项目，保证这个项目的生命力。

**Reply by l15y** [reply]
直接加入这些功能不现实，可以考虑通过webssh调用本地程序间接实现

**Reply by shmily91** [reply]
> 直接加入这些功能不现实，可以考虑通过webssh调用本地程序间接实现

刚了解了一下langchain可以实现chatpdf，langchain的功能看介绍挺强大的，也有人基于langchain做glm的知识库，但是那个项目的功能目前不如wenda完备。

**Reply by l15y** [reply]
st知识库就用了langchain

**Comment by ron21c** [comment]
[Chinese-LangChain](https://github.com/yanqiangmiffy/Chinese-LangChain) 这个项目现成的Wikipedia文本预处理语料和FAISS索引向量 是否可以引进来？

**Reply by Midnight-719** [reply]
现在st知识库就是langchain加上faiss
